### ISSUE TITLE
Exceeding an object's damage threshold causes exception

### ISSUE THREAD

**Issue by SoulSloth** [space-wizards/space-station-14#1492]
Reproduction steps:
- Hit a fuel tank, table wall, with your fists or a tool that isn't the welding torch until the damage threshold is exceeded
- Observe `DebugAssertException: Tried to get component on a deleted entity`

This is probably due to the fact that the audio system tries to get the object's location after it's marked as deleted.
```
Robust.Shared.Utility.DebugAssertException: Tried to get component on a deleted entity.
  at Robust.Shared.Utility.DebugTools.Assert(Boolean condition, String message) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\Utility\DebugTools.cs:48
  at Robust.Shared.GameObjects.Entity.TryGetComponent[T](T& component) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\Entity.cs:291
  at Robust.Shared.GameObjects.Components.Transform.TransformComponent.get_GridID() in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\Components\Transform\TransformComponent.cs:67
  at Robust.Shared.GameObjects.Components.Transform.TransformComponent.get_GridPosition() in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\Components\Transform\TransformComponent.cs:198
  at Robust.Server.GameObjects.EntitySystems.AudioSystem.PlayFromEntity(String filename, IEntity entity, Nullable`1 audioParams, Int32 range) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Server\GameObjects\EntitySystems\AudioSystem.cs:118
  at Content.Server.GameObjects.Components.Weapon.Melee.MeleeWeaponComponent.Content.Shared.Interfaces.GameObjects.Components.IAttack.Attack(AttackEventArgs eventArgs) in C:\projects\ss14\space-station-14\Content.Server\GameObjects\Components\Weapon\Melee\MeleeWeaponComponent.cs:120
  at Content.Server.GameObjects.EntitySystems.Click.InteractionSystem.DoAttack(IEntity player, GridCoordinates coordinates) in C:\projects\ss14\space-station-14\Content.Server\GameObjects\EntitySystems\Click\InteractionSystem.cs:748
  at Content.Server.GameObjects.EntitySystems.Click.InteractionSystem.HandleWideAttack(ICommonSession session, GridCoordinates coords, EntityUid uid) in C:\projects\ss14\space-station-14\Content.Server\GameObjects\EntitySystems\Click\InteractionSystem.cs:166
  at Robust.Shared.Input.Binding.PointerInputCmdHandler.<>c__DisplayClass2_0.<.ctor>b__0(PointerInputCmdArgs& args) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:92
  at Robust.Shared.Input.Binding.PointerInputCmdHandler.HandleCmdMessage(ICommonSession session, InputCmdMessage message) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:112
  at Robust.Server.GameObjects.EntitySystems.InputSystem.InputMessageHandler(InputCmdMessage message, EntitySessionEventArgs eventArgs) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Server\GameObjects\EntitySystems\InputSystem.cs:65
  at Robust.Shared.GameObjects.EventBusExt.HandlerWrapper`1.Invoke(EntitySessionMessage`1 msg) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EventBusExt.cs:23
  at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass7_0`1.<SubscribeEvent>b__0(Object ev) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.cs:151
  at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEvent(EventSource source, Object eventArgs) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.cs:285
  at Robust.Shared.GameObjects.EntityEventBus.RaiseEvent(EventSource source, Object toRaise) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.cs:201
  at Robust.Shared.GameObjects.EntityManager.<Initialize>b__22_1(Object sender, Object systemMsg) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityManager.cs:64
  at Robust.Server.GameObjects.ServerEntityNetworkManager.DispatchEntityNetworkMessage(MsgEntity message) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityNetworkManager.cs:161
  at Robust.Server.GameObjects.ServerEntityNetworkManager.HandleEntityNetworkMessage(MsgEntity message) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityNetworkManager.cs:126
  at Robust.Shared.Network.NetManager.<>c__DisplayClass85_0`1.<RegisterNetMessage>b__0(NetMessage msg) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\Network\NetManager.cs:879
  at Robust.Shared.Network.NetManager.DispatchNetMessage(NetIncomingMessage msg) in C:\projects\ss14\space-station-14\RobustToolbox\Robust.Shared\Network\NetManager.cs:826
```

**Comment by SoulSloth** [comment]
Moved the audio playing logic before the damage calculation in MeleeComponent, however objects are deleted on the client before they can play the audio.

**Comment by DrSmugleaf** [comment]
Fixed by #1494
